internal/plugin/plugin.go

```diff
@@ -0,0 +1,166 @@
+	CommandDescribeKey Command = proto.CommandDescribeKey
+
+	// CommandGenerateSignature is the name of the plugin command
+	// which must be supported by every plugin that has the
+	// SIGNATURE_GENERATOR.RAW capability.
+	CommandGenerateSignature Command = proto.CommandGenerateSignature
+
+	// CommandGenerateEnvelope is the name of the plugin command
+	// which must be supported by every plugin that has the
+	// SIGNATURE_GENERATOR.ENVELOPE capability.
+	CommandGenerateEnvelope Command = proto.CommandGenerateEnvelope
+
+	// CommandVerifySignature is the name of the plugin command
+	// which must be supported by every plugin that has
+	// any SIGNATURE_VERIFIER.* capability
+	CommandVerifySignature Command = proto.CommandVerifySignature
+)
+
+// Capability is a feature available in the plugin contract.
+type Capability = proto.Capability
+
+const (
+	// CapabilitySignatureGenerator is the name of the capability
+	// for a plugin to support generating raw signatures.
+	CapabilitySignatureGenerator Capability = proto.CapabilitySignatureGenerator
+
+	// CapabilityEnvelopeGenerator is the name of the capability
+	// for a plugin to support generating envelope signatures.
+	CapabilityEnvelopeGenerator Capability = proto.CapabilityEnvelopeGenerator
+
+	// CapabilityTrustedIdentityVerifier is the name of the capability
+	// for a plugin to support verifying trusted identities.
+	CapabilityTrustedIdentityVerifier = Capability(VerificationCapabilityTrustedIdentity)
+
+	// CapabilityRevocationCheckVerifier is the name of the capability
+	// for a plugin to support verifying revocation checks.
+	CapabilityRevocationCheckVerifier = Capability(VerificationCapabilityRevocationCheck)
+)
+
+// VerificationCapability is a verification feature available in the plugin contract.
+type VerificationCapability string
+
+const (
+	// VerificationCapabilityTrustedIdentity is the name of the capability
+	// for a plugin to support verifying trusted identities.
+	VerificationCapabilityTrustedIdentity VerificationCapability = VerificationCapability(proto.CapabilityTrustedIdentityVerifier)
+
+	// VerificationCapabilityRevocationCheck is the name of the capability
+	// for a plugin to support verifying revocation checks.
+	VerificationCapabilityRevocationCheck VerificationCapability = VerificationCapability(proto.CapabilityRevocationCheckVerifier)
+)
+
+// SigningScheme formalizes the feature set provided by the signature produced using a signing scheme
+type SigningScheme = proto.SigningScheme
+
+const (
+	// SigningSchemeDefault defines a signing scheme that uses the traditional signing workflow
+	// in which an end user generates signatures using X.509 certificates
+	SigningSchemeDefault SigningScheme = proto.SigningSchemeDefault
+
+	// SigningSchemeAuthority defines a signing scheme in which a signing authority
+	// generates signatures on behalf of an end user using X.509 certificates
+	SigningSchemeAuthority SigningScheme = proto.SigningSchemeAuthority
+)
```

Comment:
These are already defined in [notation-core-go](https://github.com/notaryproject/notation-core-go/blob/main/signature/types.go#L20). Do we really need them in `plugin` package?

Reply:
Removed. Also added an issue for the spec: https://github.com/notaryproject/notaryproject/issues/200